### ISSUE TITLE
docs: filtering embedded values

### ISSUE THREAD

**Issue by clement-estone** [mikro-orm/mikro-orm#2891]
Hi there!

I've been using Mikro ORM for quite a bit now and I'm really enjoying it so far. I only have one problem which is the following:

I have an entity (let's call it `User`) with an embedded property (let's call it `UserData`). The latter contains a field that can either hold a value or be `null`.

Now, I have to search through all the `User` entities to find the ones that have this specific field not set to `null`, which gives something like:

```
this.userRepository.find({ userData: { theNullableField: { $ne: null } } }) // Nest.js code, using an EntityRepository<User>
```

This compiles correctly but at runtime gives me the following error:

```
Using operators inside embeddables is not allowed, move the operator above.
```

So my question is: how can I make this search? I didn't find any workaround in the docs or in the issues that would help me with this, so I'd like to know how this can be done (what does `move the operator above` means? from what I understand operators can only be applied to the current value and not to sub-values, at least that's what's indicated in the docs).

I'm using MikroORM v5.0.5.

Thanks in advance for your help!

**Comment by B4nan** [comment]
Sounds like the validation is too aggressive, it should disallow operators "in the middle" of the query, e.g. `{ prop: { $lt: { nested: 1 } }`, which won't work for embeddables, as `prop.nested` is actually one property.

Can you provide more details like entity definition? Just tried this in the tests and I am not able to reproduce.

**Comment by clement-estone** [comment]
Here is a minimal reproduction code:

```
@Embeddable()
export class UserData {
  @Property()
  profileUri?: string
}

@Entity()
export class User {
  @Embedded()
  data!: UserData
}

@Injectable()
export class UserService {
  constructor(private readonly userRepo: EntityRepository<User>) {}

  async getUsersWithProfileUri(): Promise<User[]> {
    return this.userRepo.find({
      data: {
        $ne: null
      }
    })
  }
}
```

**Comment by B4nan** [comment]
Unless you use ts-morph, this code is not valid, you need to specify the type in `@Embedded` decorator. 

Also, you are using inline or object embeddables? What driver?

**Comment by clement-estone** [comment]
I am indeed using TS-Morph, without it this code would throw an error on startup with MikroORM saying it could not resolve the type of this field.

I'm using the PostgreSQL driver.

I suppose the embeddables are used in inline mode? Each property of the embeddable has its own dedicated column in the parent entity's database table. I can't really grasp the difference between inline and object from the docs, but the code I've pasted here is exactly how I embed the entities, there is no additional property provided to the decorators.

**Comment by B4nan** [comment]
Ok that might be the difference, I was testing with object embeddable. FYI the difference is that inline mode means embedded properties will live separately on the owning entity, while object embeddable is a single JSON column. I guess we should make the docs clearer and add some examples.

**Comment by clement-estone** [comment]
> Ok that might be the difference, I was testing with object embeddable. FYI the difference is that inline mode means embedded properties will live separately on the owning entity, while object embeddable is a single JSON column. I guess we should make the docs clearer and add some examples.

So how can I make use the operator in these conditions? What syntax should I use to perform the search?

**Comment by B4nan** [comment]
I still can't reproduce this, it works fine on my end. Please provide executable reproduction, ideally as a test case

https://github.com/mikro-orm/mikro-orm/tree/master/tests/issues

**Comment by clement-estone** [comment]
I just tried on another computer, and it seems to work perfectly fine! I can't get why, I removed the `node_modules` to reinstall it (using a lockfile), removed the entire database etc. but it only works on one machine.

I'll dig a bit into that to see where that comes from.

**Comment by B4nan** [comment]
If you use ts-morph, you also have metadata cache in `temp`, be sure to wipe that too. It should get invalidated automatically on entity changes tho.

**Comment by clement-estone** [comment]
> If you use ts-morph, you also have metadata cache in `temp`, be sure to wipe that too. It should get invalidated automatically on entity changes tho.

Yes I already did that. I can't get why this only works on one computer and not the other, as they are using the same operating system, same environment, same configuration. Very weird. I'll try to see if I can find an explanation on my side.

**Comment by B4nan** [comment]
Ok, will move this to discussions, this really sounds like your enviornment is somehow misconfigured.
